docs/architecture/index.rst

```diff
@@ -23,3 +23,74 @@ agent has no state and only communicates with the master. Each agent is responsi
 The **trial runner** runs a trial in a containerized environment. So the trial runners are expected
```

Comment:
I still don't understand exactly what a "trial runner" describes? Is it the TrialController in Trial classes?

It is strange that we mention the "trial runner" but do not show it on any of the diagrams. It seems like it would make more sense to have bullet points for the task containers, workers, and model code rather than the "trial runner." However, I realize that perhaps there is something I don't understand here.

Reply:
I think "Worker" in the diagram would be the Trial Runner in this case. I will dig into the code and rename it.

Reply:
Isn't the "worker" just a process exec'd from the container entrypoint? I feel like it would add to this document to explain that many processes may exist inside of a single container. Or perhaps that is out of scope of this?

Reply:
"trial runner" is a historic phrase for the container in which a trial runs.  Historic as in "from the time before we had notebooks or tensorboards or shells or command".

Reply:
(in other words, these docs haven't seen love in a _long_ time)